RandLAPACK/comps/rl_util.hh

```diff
@@ -477,6 +678,30 @@ void gen_mat_type(
                 RandBLAS::dense::fill_buff(A_dat, D, state);
             }
             break;
+        case 7: {
+                // Generating matrix with a staircase-like spectrum
+                RandLAPACK::util::gen_step_mat(m, n, A, k, std::get<1>(type), std::get<2>(type), state);
+            }    
+            break;
+        case 8: {
+                // This matrix may be numerically rank deficient
+                RandLAPACK::util::gen_spiked_mat(m, n, A, state);
+                if(std::get<2>(type))
+                    k = rank_check(m, n, A);
```

Comment:
``rank_check`` doesn't raise any errors. I also don't see where you use ``k`` later on. So it's not clear to me what the rank check accomplishes here.

Reply:
Ah, didn't see that k was provided by the user here.